ddl/table.go

```diff
@@ -679,6 +686,15 @@ func onTruncateTable(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64, _ erro
 		job.State = model.JobStateCancelled
 		return ver, infoschema.ErrTableNotExists.GenWithStackByArgs(job.SchemaName, tblInfo.Name.O)
 	}
+	referredFK, err := checkTableHasForeignKeyReferredInOwner(d, t, job.SchemaName, job.TableName, []ast.Ident{{Name: tblInfo.Name, Schema: model.NewCIStr(job.SchemaName)}}, fkCheck)
+	if err != nil {
+		return ver, err
+	}
+	if referredFK != nil {
+		job.State = model.JobStateCancelled
+		msg := fmt.Sprintf("`%s`.`%s` CONSTRAINT `%s`", referredFK.ChildSchema, referredFK.ChildTable, referredFK.ChildFKName)
+		return ver, errors.Trace(dbterror.ErrTruncateIllegalForeignKey.GenWithStackByArgs(msg))
```

Comment:
Why not put these lines to https://github.com/pingcap/tidb/pull/37773/files#diff-2218b0b129e90b1e049edf99a3a1c5192249feadb02f54a00b426bf14ac9d71dR338？Unify `checkDropTableHasForeignKeyReferredInOwner`.

Reply:
great, done.